server/src/main/java/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -971,6 +988,13 @@ public List<IPAddressVO> doInTransaction(TransactionStatus status) throws Insuff
 
                 if (lockOneRow) {
                     assert (addrs.size() == 1) : "Return size is incorrect: " + addrs.size();
+                    IpAddress ipAddress = addrs.get(0);
+                    boolean ipCanBeAllocated = checkIfPublicIpAddressIsNotInQuarantineAndCanBeAllocated(ipAddress, owner);
+
+                    if (!ipCanBeAllocated) {
+                        throw new InsufficientAddressCapacityException(String.format("Failed to allocate public IP address [%s] as it is in quarantine.", ipAddress.getAddress()),
+                                DataCenter.class, dcId);
+                    }
```

Comment:
I think it would be best to remove/replace the assert and extract this to a private method

Reply:
Hey @DaanHoogland, most of the time I would agree with this, even though the method would only be called once, there are scenarios that would improve code legibility. However,  in this case, I think the code is already clear.